### DISCUSSION TITLE
Deploy to github pages with custom domain not working

### DISCUSSION THREAD

**Discussion by shibumi** [nunocoracao/blowfish#276] [Q&A]
**Describe the bug**
We have deployed a new website (https://github.com/sre-muc/sremuc) with the blowfish theme.
The overall github pages configuration seems to be correct, but for some reason the path for all javascript files is wrong.

The page should be clearly visible on: https://muc.sre.xyz/
but when I look into the debugging tools I see a domain such like this one:

```
https://muc.sre.xyz/muc.sre.xyz/css/main.bundle.min.b341e75d494bde9b8b3f27689cff89cd80ec20878b291b6fb53197b0d0ba85f9ad1a26bd179879742cabb98b9f3cff7fa71cacca60994d398dc9e629e00596be.css
```

The interesting part is: https://muc.sre.xyz/muc.sre.xyz/.... 

is this some caching problem? The base URL seems to be wrong, but the configuration should be correct now.

**To Reproduce**
1. Fork our website https://github.com/sre-muc/sremuc
2. Setup gh-pages
3. See the duplicated path in the base URL only for javascript files

**Expected behavior**
The website should work out of the box

**Hugo & Blowfish versions**
Hugo: '0.105.0'

**Comment by nunocoracao** [comment]
Hello @shibumi this is not a blowfish issue... might be a problem with configuration on your side. 

I checked:

1. Your config file seems to be correct and pointing to https://sre-muc.github.io/sremuc/
2. The site in https://sre-muc.github.io/sremuc/ seems to be displaying everything properly
3. https://muc.sre.xyz/ is pulling the files using the relative path hence the problem

Not sure how you are configuring the DNS part of this, but if you change the base url config to the final site everything should work. The github pages one will break. Nothing here is related with the blowfish theme. You are trying to support two different deploys with the same URL.

**Comment by shibumi** [comment]
Actually I tried that before. I tried adding the custom domain as baseURL
and putting a CNAME file in the gh-pages branch root directory.

The result is that the page at sre-muc.github.io/sremuc/ will break and the
new page at muc.sre.xyz will still be broken, because for some unknown
reason the index.html file will have wrong baseURLs.

If I set the following baseURL: "https://muc.sre.xyz"  the index.html file
will inject "muc.sre.xyz" as path, hence I will run into wrong URLs like "
https://muc.sre.xyz/muc.sre.xyz/.../..ja

This doesn't seem to be Hugo or GitHub pages related for me, because many
other GitHub pages work fine with custom domain and Hugo.


Did ever try running your theme with a custom domain via gh-pages?

**Comment by shibumi** [comment]
Please checkout the git commit history on the master branch of
GitHub.com/sre-muc/sremuc
You will see that I tried different constellations of baseURL and all of
that failed.

**Comment by nunocoracao** [comment]
Not sure I can do much here. Is there a way for me to check those DNS configs? 

From what I can see https://sre-muc.github.io/sremuc/ is working so all good with the base_url if you want to serve the site from a different URL it should also be fine. I am not sure what it's happening there. 

Two ideas:

1. look at my github actions... maybe there is something different for you to try
2. did you check https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site ?

you can check my repos and see all my different configs. My homepage is on github pages and in firebase with a custom domain (using different repos). Never try to do something like that nor am I really sure how it works.

**Comment by nunocoracao** [comment]
Move this into discussions to check if someone can help

**Comment by shibumi** [comment]
@nunocoracao  do you have any idea why the "show more" button gives us a 404, even with everything configured correctly in our repository? https://sre-muc.github.io/sremuc/

I am still 100% sure something is either wrong with the theme or github pages is just  buggy as hell.

**Reply by shibumi** [reply]
The show more button seems to miss the directory suffix: https://sre-muc.github.io/events. This works: https://sre-muc.github.io/sremuc/events/. But why? Shouldn't this be configured through the config file?  See also: https://github.com/SRE-MUC/sremuc/blob/master/config/_default/config.toml#L5

**Comment by nunocoracao** [comment]
showMoreLinkDest = "/events" in the params.toml -> add the /sremuc/events
  The path is global to the website... if the website is in a specific path like your case you need to add it to the link of the "show more" button

**Reply by nunocoracao** [reply]
@shibumi did it work?

**Reply by shibumi** [reply]
Hi @nunocoracao  yes, this worked. Thank you so much. We just still run into this problem with the custom domain. I really don't understand why switching to a custom domain does not work.

Can you test in anyway if your theme really works with a custom domain or can you point to an example website, where it works?
I really think this is either a theme issue, configuration issue or github pages issue.

**Reply by nunocoracao** [reply]
Just did the same thing to Blowfish. You need two diff hosting locations for each URL if you are planning to use a path for deploying one of them. If both were in the root it would work but that is not the case for you. Check blowfish github actions, deploying to github pages and firebase. Hope this helps

**Reply by shibumi** [reply]
Thank you for your answer. Did you try GitHub pages with custom domain by accident? I would really like to know if it's an issue with Blowfish or with Github pages.

I am not interested in running two different setups, I would be happy already if the page would work with our custom domain, but last time when I tried it, it didn't work :/

Now I am confused if it's an issue with GitHub pages or something blowfish specific.

**Reply by nunocoracao** [reply]
@shibumi nothing wrong with neither of them. Hugo requires a base URL to generate the static pages: you can't have a single deploy with different base URLs which are not both at the root (or in the same path). It's very simple.
